internal/wasi/fs.go

```diff
@@ -5,100 +5,137 @@ import (
 	"io"
 	"io/fs"
 	"os"
+	"path"
 	"runtime"
+	"sort"
 	"strings"
-
-	"github.com/tetratelabs/wazero/wasi"
+	"time"
 )
 
+// DirFS is a file system that implements fs.FS and wasi.OpenFileFS.
```

Comment:
do you know any go api for adding a file? maybe there's an open, closed or locked issue on go's repo if not.

Reply:
I also investigated whether there is any Go's writable FS interface. There is an issue discussing creating/writing file on FS, but it lacks the attention from the Go team. https://github.com/golang/go/issues/45757

Reply:
ah, or if your intention is literally API to add a new file, it's exactly `os.OpenFile`, which also can do file creation. `os.Create` is simpler API that creates a file with default flags. It does `OpenFile(name, O_RDWR|O_CREATE|O_TRUNC, 0666)` internally.